src/components/themed/SimpleTextInput.tsx

```diff
@@ -148,6 +154,18 @@ export const SimpleTextInput = React.forwardRef<SimpleTextInputRef, SimpleTextIn
     if (inputRef.current != null) inputRef.current.setNativeProps(nativeProps)
   }
 
+  // This fixes RN bugs with controlled TextInput components.
+  // By avoiding the `value` prop for the TextInput, we can avoid the bugs.
+  // So an alternative to using the `value` prop, is to imperatively set the,
+  // TextInput's text value using `setNativeProps({ text: value })`.
+  // We do this only if the `value` prop from this component has changed.
+  React.useEffect(() => {
+    if (inputRef.current != null && value !== valueRef.current) {
+      valueRef.current = value
+      inputRef.current.setNativeProps({ text: value })
+    }
+  }, [inputRef, value])
```

Comment:
Since the `inputRef` is a reference, it shouldn't be in the deps list.

Reply:
tell that to @react-hooks eslint plugin 🤷‍♂️

Reply:
Weird, but OK.